### ISSUE TITLE
gettext on 10.10.4: configure: error: cannot run C compiled programs

### ISSUE THREAD

**Issue by wiml** [Homebrew/homebrew#45198]
I can't build `gettext` (or `libao`). It produces an error along these lines:

```
==> ./configure --disable-silent-rules --prefix=/Volumes/Local/Homebrew/Cellar/gettext/0.19.6 --
Last 15 lines from /Users/wimlocal/Library/Logs/Homebrew/gettext/01.configure:
checking for gawk... no
checking for mawk... no
checking for nawk... no
checking for awk... awk
checking whether make sets $(MAKE)... yes
checking whether make supports nested variables... yes
checking for gcc... clang
checking whether the C compiler works... yes
checking for C compiler default output file name... a.out
checking for suffix of executables... 
checking whether we are cross compiling... configure: error: in `/Volumes/Local/tmp/gettext20151020-19780-1pyqrv2/gettext-0.19.6/gettext-runtime':
configure: error: cannot run C compiled programs.
If you meant to cross compile, use `--host'.
See `config.log' for more details
configure: error: ./configure failed for gettext-runtime
```

Oddly, if I use `brew install gettext --debug`, enter the shell, and copy&paste the same configure command, it runs without any errors. The same problem happens during the make and make-install steps: brew's invocation fails, but if I let brew open a subshell and type the same command that brew was executing, it succeeds. So this way I successfully installed gettext and libao, semi-manually.

[Gist-logs are here](https://gist.github.com/anonymous/5476e86a4dbed2050b0e)

**Comment by DomT4** [comment]
If you're running Xcode 7 on Yosemite you need to install the Command Line Tools to get around `sysroot` being set to null on account of Xcode 7 not having Yosemite's SDK.

**Comment by wiml** [comment]
Huh, "installing the command line tools" is different from installing Xcode and selecting that xcode's command line tools? Unfortunately `xcode-select --install` fails for me "because of a network problem".

However, xcode-selecting an older version of Xcode (I had Xcode 6.2 lying around) also allowed homebrew to work again.

**Comment by mikemcquaid** [comment]
> Huh, "installing the command line tools" is different from installing Xcode and selecting that xcode's command line tools? Unfortunately xcode-select --install fails for me "because of a network problem".

Yes. Unfortunately you'll need to just retry this a few times.

**Comment by DomT4** [comment]
Closing per the above. If installing the CLT doesn't resolve the problem, please feel free to let us know. Apologies there's not much we can do aside work around Apple not including the 10.10 SDK in Xcode 7 :cry:.

**Comment by mikemcquaid** [comment]
> Closing per the above. If installing the CLT doesn't resolve the problem, please feel free to let us know. Apologies there's not much we can do aside work around Apple not including the 10.10 SDK in Xcode 7 .

Still the case in Xcode 7.1?

**Comment by DomT4** [comment]
There's still no 10.10 SDK in the Xcode 7.1 builds AFAICS, despite the 2.2GB size hump with the latest Xcode.

**Comment by mikemcquaid** [comment]
@DomT4 I wonder if we need to work out some other solution then.

**Comment by DomT4** [comment]
@mikemcquaid I'm open to suggestions. I don't think nudging people to install the small CLT package isn't too awful a workaround though. Part of me wonders if Apple is actively trying to dump people off 10.10 with this move.

**Comment by mikemcquaid** [comment]
@DomT4 Maybe we could make superenv fix 10.10 SDK references or something? @xu-cheng, any thoughts?
